lib/sqlalchemy/sql/compiler.py

```diff
@@ -3610,8 +3661,9 @@ def visit_insert(self, insert_stmt, **kw):
         if returning_clause and not self.returning_precedes_values:
             text += " " + returning_clause
 
-        if self.ctes and toplevel and not self.dialect.cte_follows_insert:
-            text = self._render_cte_clause() + text
+        if self.ctes and not self.dialect.cte_follows_insert:
```

Comment:
in this case, the "cte_follows_insert" flag actually has us make a different choice as to how to render, so that flag is more useful